From 98b8f83febd45be97dc1b4583594ee5d555526ad Mon Sep 17 00:00:00 2001
From: Hari Chandrakanthan <quic_haric@quicinc.com>
Date: Tue, 24 Jan 2023 14:55:31 +0530
Subject: [PATCH] ath12k : fix sta association in mbssid case

ath12k station is not able to associate to non-transmitting vif of mbss.
This is because wpa_supplicant does not receive the information about
non-transmitting vif as part of NL scan results.

ath12k driver does not advertises its mbssid capability to mac80211.
So wiphy->support_mbssid is not set.

Since support_mbssid is not set, the information about non-transmitting vif
is not parsed from the rx mgmt frame and it is not added as part of the scan
results.

Fix this by advertising the MBSSID capabitlity of ath12k driver to mac80211.

Signed-off-by: Hari Chandrakanthan <quic_haric@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index 0bd4f92..f30403e 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -13920,6 +13920,7 @@ static int ath12k_mac_hw_register(struct ath12k_hw *ah)
 	ieee80211_hw_set(hw, SUPPORTS_AMSDU_IN_AMPDU);
 	ieee80211_hw_set(hw, USES_RSS);
 	ieee80211_hw_set(hw, SUPPORTS_TID_CLASS_OFFLOAD);
+	ieee80211_hw_set(hw, SUPPORTS_MULTI_BSSID);
 
 	if (ath12k_frame_mode == ATH12K_HW_TXRX_ETHERNET) {
 		ieee80211_hw_set(hw, SUPPORTS_TX_ENCAP_OFFLOAD);
-- 
2.7.4

